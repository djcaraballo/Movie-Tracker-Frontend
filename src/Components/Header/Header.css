.header {
  margin-bottom: 5rem;
}

.fa-sign-out-alt,
.fa-sign-in-alt {
  color: #0496a0;
  font-size: 1.5em;
  display: block;
  text-align: right;
  padding: 2em;
}

.fa-sign-out-alt:hover,
.fa-sign-in-alt:hover {
  color: #fff;
  cursor: pointer;
}

.app-title {
  background: -webkit-gradient(
    linear,
    left top,
    left bottom,
    from(#ecf0ef),
    to(#066168)
  );
  -webkit-background-clip: text;
  background-clip: text;
  -webkit-text-fill-color: transparent;
  display: inline-block;
  font-family: "Montserrat", sans-serif;
  font-size: 7em;
  font-weight: 900;
  letter-spacing: 5px;
  position: relative;
  text-transform: uppercase;
  text-shadow: 6px 0 #01838c;
  border-bottom: #01838c double;
}

.app-title::before {
  background-position: -180px;
  -webkit-animation: flare 5s infinite;
  animation: flare 5s infinite;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
  background-image: linear-gradient(
    65deg,
    transparent 20%,
    rgba(255, 255, 255, 0.2) 20%,
    rgba(255, 255, 255, 0.3) 27%,
    transparent 27%,
    transparent 100%
  );
  -webkit-background-clip: text;
  background-clip: text;
  -webkit-text-fill-color: transparent;
  content: "MovieTracker";
  color: #fff;
  display: block;
  padding-right: 140px;
  position: absolute;
}

.sparkle::after,
.sparkle::before {
  -webkit-animation: sparkle 5s infinite;
  animation: sparkle 5s infinite;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
  background: #fff;
  border-radius: 100%;
  box-shadow: 0 0 5px #fff, 0 0 10px #fff, 0 0 15px #fff, 0 0 20px #fff,
    0 0 25px #fff, 0 0 30px #fff, 0 0 35px #fff;
  content: "";
  display: block;
  height: 10px;
  opacity: 0.7;
  position: absolute;
  width: 10px;
}

.sparkle::before {
  -webkit-animation-delay: 0.2s;
  animation-delay: 0.2s;
  height: 7px;
  left: 0.1em;
  top: 0.9em;
  width: 7px;
}

.sparkle::after {
  top: 0.32em;
  right: 0px;
}

@-webkit-keyframes flare {
  0% {
    background-position: -180px;
  }
  30% {
    background-position: 700px;
  }
  100% {
    background-position: 700px;
  }
}

@keyframes flare {
  0% {
    background-position: -180px;
  }
  30% {
    background-position: 700px;
  }
  100% {
    background-position: 700px;
  }
}

@-webkit-keyframes sparkle {
  0% {
    opacity: 0;
  }
  30% {
    opacity: 0;
  }
  40% {
    opacity: 0.8;
  }
  60% {
    opacity: 0;
  }
  100% {
    opacity: 0;
  }
}

@keyframes sparkle {
  0% {
    opacity: 0;
  }
  30% {
    opacity: 0;
  }
  40% {
    opacity: 0.8;
  }
  60% {
    opacity: 0;
  }
  100% {
    opacity: 0;
  }
}

@media (max-width: 1050px) {
  .app-title {
    font-size: 6.5em;
  }
}

@media (max-width: 975px) {
  .app-title {
    font-size: 5.75em;
    text-shadow: 5px 0 #01838c;
  }
}

@media (max-width: 875px) {
  .app-title {
    font-size: 5em;
  }
}

@media (max-width: 775px) {
  .app-title {
    font-size: 4em;
    text-shadow: 4px 0 #01838c;
  }
}

@media (max-width: 675px) {
  .app-title {
    font-size: 3.5em;
    text-shadow: 3px 0 #01838c;
  }

  .header {
    margin-bottom: 2rem;
  }
}

@media (max-width: 575px) {
  .app-title {
    font-size: 3em;
    text-shadow: 3px 0 #01838c;
  }
}

@media (max-width: 475px) {
  .app-title {
    font-size: 2em;
  }

  .header {
    margin-bottom: 1rem;
  }
}
