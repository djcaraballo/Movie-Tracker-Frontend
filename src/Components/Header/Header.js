import React from "react";
import { signOutUser } from "../../Actions/user-actions";
import { NavLink } from "react-router-dom";
import { connect } from "react-redux";
import "./Header.css";

export const Header = (props, signOutUser) => {
  return (
    <div className="header">
      <span>
        {props.user ? (
          <NavLink to="/">
            <i
              onClick={() => props.signOutUser(props.user)}
              className="fas fa-sign-out-alt"
            />
          </NavLink>
        ) : (
          <NavLink to="/login">
            <i className="fas fa-sign-in-alt" />
          </NavLink>
        )}
      </span>
      <span className="app-title">
        <span className="sparkle">MovieTracker</span>
      </span>
    </div>
  );
};

export const mapStateToProps = state => ({
  user: state.user
});

export const mapDispatchToProps = dispatch => ({
  signOutUser: user => dispatch(signOutUser(user))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Header);
