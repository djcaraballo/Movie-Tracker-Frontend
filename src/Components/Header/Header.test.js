import React from 'react';
import ReactDOM from 'react-dom';
import { Header, mapStateToProps, mapDispatchToProps } from './Header';
import { shallow } from 'enzyme';

describe('Header Component', () => {
  let wrapper;
  const mockUser = {
    name: 'Louisa',
    id: 1806,
    email: 'Louisa@turing.io',
    password: 'oooooooooooookay'
  };
  const mockSignOutUser = jest.fn();

  it('should match the snapshot', () => {
    wrapper = shallow(
      <Header
        user={mockUser}
        signOutUser={mockSignOutUser}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  describe('mapStateToProps function', () => {
    it('should return an object with a user', () => {
      const mockState = {
        movies: [
          {title: 'Coco', vote_average: 7.6, favorite: true},
          {title: 'Inception', vote_average: 8.6, favorite: true},
          {title: 'Hunt for Red October', vote_average: 4.6, favorite: true}
        ],
        user: {name: 'Louisa', email: 'blah@gmail.com'},
        favorites: []
      }
      const expected = {user: {name: 'Louisa', email: 'blah@gmail.com'}}
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })

  describe('mapDispatchToProps function', () => {

  })

})