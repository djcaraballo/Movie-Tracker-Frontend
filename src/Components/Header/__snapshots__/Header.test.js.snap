// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    signOutUser={[MockFunction]}
    user={
      Object {
        "email": "Louisa@turing.io",
        "id": 1806,
        "name": "Louisa",
        "password": "oooooooooooookay",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span>
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/"
          >
            <i
              className="fas fa-sign-out-alt"
              onClick={[Function]}
            />
          </NavLink>
        </span>,
        <span
          className="app-title"
        >
          <span
            className="sparkle"
          >
            MovieTracker
          </span>
        </span>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            aria-current="page"
            to="/"
          >
            <i
              className="fas fa-sign-out-alt"
              onClick={[Function]}
            />
          </NavLink>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": <i
              className="fas fa-sign-out-alt"
              onClick={[Function]}
            />,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-sign-out-alt",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": [Function],
        },
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span
            className="sparkle"
          >
            MovieTracker
          </span>,
          "className": "app-title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "MovieTracker",
            "className": "sparkle",
          },
          "ref": null,
          "rendered": "MovieTracker",
          "type": "span",
        },
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span>
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/"
            >
              <i
                className="fas fa-sign-out-alt"
                onClick={[Function]}
              />
            </NavLink>
          </span>,
          <span
            className="app-title"
          >
            <span
              className="sparkle"
            >
              MovieTracker
            </span>
          </span>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              aria-current="page"
              to="/"
            >
              <i
                className="fas fa-sign-out-alt"
                onClick={[Function]}
              />
            </NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": <i
                className="fas fa-sign-out-alt"
                onClick={[Function]}
              />,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-sign-out-alt",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": [Function],
          },
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="sparkle"
            >
              MovieTracker
            </span>,
            "className": "app-title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "MovieTracker",
              "className": "sparkle",
            },
            "ref": null,
            "rendered": "MovieTracker",
            "type": "span",
          },
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
